src/plugins/common.go

```diff
@@ -0,0 +1,109 @@
+package plugins
+
+import (
+	"github.com/nginx/agent/sdk/v2/client"
+	"github.com/nginx/agent/v2/src/core"
+	"github.com/nginx/agent/v2/src/core/config"
+	"github.com/nginx/agent/v2/src/extensions"
+	log "github.com/sirupsen/logrus"
+
+	agent_config "github.com/nginx/agent/sdk/v2/agent/config"
+
+	sdkGRPC "github.com/nginx/agent/sdk/v2/grpc"
+
+	"github.com/google/uuid"
+)
+
+func LoadPlugins(commander client.Commander, binary core.NginxBinary, env core.Environment, reporter client.MetricReporter, loadedConfig *config.Config) ([]core.Plugin, []core.ExtensionPlugin) {
+	var corePlugins []core.Plugin
+	var extensionPlugins []core.ExtensionPlugin
+
+	if commander != nil {
+		corePlugins = append(corePlugins,
+			NewCommander(commander, loadedConfig),
+		)
+
+		if loadedConfig.IsFeatureEnabled(agent_config.FeatureFileWatcher) {
+			corePlugins = append(corePlugins,
+				NewFileWatcher(loadedConfig, env),
+				NewFileWatchThrottle(),
+			)
+		}
+	}
+
+	if loadedConfig.IsFeatureEnabled(agent_config.FeatureMetrics) || loadedConfig.IsFeatureEnabled(agent_config.FeatureMetricsSender) && reporter != nil {
```

Comment:
Lets ensure the fix from https://github.com/nginx/agent/pull/465 is carried over:
```suggestion
	if (loadedConfig.IsFeatureEnabled(agent_config.FeatureMetrics) || loadedConfig.IsFeatureEnabled(agent_config.FeatureMetricsSender)) && reporter != nil {

```